QgisModelBaker/libqgsprojectgen/dataobjects/layers.py

```diff
@@ -186,3 +212,27 @@ def real_id(self):
             return self.__layer.id()
         else:
             return None
+
+    def isPureLinkTable(self, project):
+        '''
+        Returns True if the layer is a pure link table in a n:m relation.
+        With "pure" it is meant the layer has no more fields than foreign keys and its id.
+        '''
+
+        # Check if table is a link table
+        if self.is_nmrel is False:
```

Comment:
I usually prefer `if not self.is_nmrel` - but not sure about any convention about it.

Reply:
Again habit from c++ where I don't like writing `!is_nmrel` because the `!` is easily overseen. But the `not` is two more characters so I can accept it ;)